### PR TITLE
feat: add channel close feature

### LINKED ISSUE #369: [feat] implement MsgChannelCloseInit and MsgChannelCloseConfirm messages both in Axon and CKB
Axon and CKB endpoints don't support close channel, this feature isn't emergency but is important, mark it here

### LINKED ISSUE #381: Channel close feature of CKB endpoint

### PR DESCRIPTION
**Pull request by ashuralyk**
## Related issues
- Closes #381 
- Resolves #369 

## Description
Channel layer misses **close feature** for a long time, and for a compatible version of IBC protocol, `close feature` should be accomplished at last.

## Change logs
0. bump [ckb-ics](https://github.com/synapseweb3/ckb-ics.git) to https://github.com/synapseweb3/ckb-ics/commits/0299a9126a7cb28ff54b1a02f286f0618133731c
1. bump forcerelay-ckb-sdk to https://github.com/synapseweb3/forcerelay-ckb-sdk/commits/cd29a30b350ab853a496a58b970017bfa1249182
3. complete transaction assembly for MsgChannelCloseInit and MsgChannelCloseConfirm
4. iterate contracts of IBC for CKB endpoint
5. extract the envelope struct from IBC transaction to distinguish which type of message is behind a closed channel, specifically to distinguish a channel that comes to close is triggered by MsgChannelCloseInit or MsgChannelCloseConfirm
6. emit channel close command in packet related CI tests
7. complete `query_txs()` method, but others are unreachable to do
   > Scan and check the incomplete methods in Ckb4Ibc, and find only `query_txs()` needs to implemented, others can keep unimplemented status.
8. add channel close trigger in CI test only between CKB chains

## Related PRs
- https://github.com/synapseweb3/ckb-ics/pull/21
- https://github.com/synapseweb3/forcerelay-ckb-contracts/pull/31
- https://github.com/synapseweb3/forcerelay-ckb-sdk/pull/39

### CODE REVIEW

**Review by jjyr — APPROVED** [review]

**Comment by ashuralyk** [comment]
@blckngm this PR's CI won't pass because we give relayer a random key but the module is set `PRIVKEY`, which would emit error in the contract check of MsgChannelCloseInit, we should send transaction to close channel via `forcerelay-ckb-sdk`

**Comment by ashuralyk** [comment]
@Flouse this PR can start to review

**Review by jjyr — APPROVED** [review]

File: crates/relayer/src/chain/ckb4ibc.rs [hunk 1]
```diff
@@ -1273,9 +1275,62 @@ impl ChainEndpoint for Ckb4IbcChain {
...
+                let block = self.rt.block_on(
+                    self.rpc_client
+                        .get_block_by_number(consensus_height.revision_height().into()),
+                )?;
+                let block_number: u64 = block.header.inner.number.into();
+                block
+                    .transactions
+                    .into_iter()
+                    .map(|tx| {
+                        let event = transaction_to_event(&tx, &prefix)?;
```
Comment (Flouse) [inline crates/relayer/src/chain/ckb4ibc.rs:1296]: If the CKB block contains `non-IBC-event` transactions, then it may fail here.

Comment (ashuralyk) [reply crates/relayer/src/chain/ckb4ibc.rs:1296]: done, review again

File: tools/ibc-test/Cargo.toml [hunk 2]
```diff
@@ -45,7 +45,7 @@ toml_edit = "0.19.14"
 lazy_static = "1.4.0"
 ethers = { version = "2.0.2", features = ["rustls", "ws"] }
 
-forcerelay-ckb-sdk = { git = "https://github.com/synapseweb3/forcerelay-ckb-sdk", rev = "64978b97" }
+forcerelay-ckb-sdk = { git = "https://github.com/synapseweb3/forcerelay-ckb-sdk", rev = "61a78aa" }
```
Comment (Flouse) [inline tools/ibc-test/Cargo.toml]:
```suggestion
forcerelay-ckb-sdk = { git = "https://github.com/synapseweb3/forcerelay-ckb-sdk", rev = "cd29a30b350ab853a496a58b970017bfa1249182" }
```

Comment (Flouse) [reply tools/ibc-test/Cargo.toml]: fix: unable to update https://github.com/synapseweb3/forcerelay-ckb-sdk?rev=61a78aa#61a78aaa

See https://github.com/synapseweb3/forcerelay/actions/runs/6953138151/job/18917831578#step:5:49

**Review by Flouse — APPROVED** [review]
LGTM

**Pull request merged**